### PR TITLE
Add options to discover initial color mode from URL param

### LINKED ISSUE #78: Discover initial color mode from URL parameter
Hi. I was wondering if there is some way to standarize how to obtain the initial color mode when using `use_color_mode`. Currently, I'm getting it from:

1. The URL parameter `color-scheme`. If exists, is updated on localstorage and used as the initial color mode.
2. Otherwise try to get from the localstorage key `color-scheme`.

Seems to me that it would not be crazy to implement this logic in `use_color_mode` directly. Perhaps adding the next parameters is enough:

- `initial_value_from_url` (_bool_): Try to discover the initial value from a URL parameter.
- `initial_value_from_url_param` (_str_): URL parameter to use trying to discover the initial value.
- `initial_value_from_storage` (_bool_): Try to discover the initial value from storage.

It the color mode is not found trying to discover it following this order would fallback to `initial_value` and `use_preferred_dark`. What do you think about it?

Perhaps a more general API to set autodiscovered values in all functions that accept an `initial_value` would be better.

### PR DESCRIPTION
**Pull request by mondeja**
Closes #78

I'm not sure if this is the best way to implement it, but seems to work fine. Suggestions would be appreciated.

### CODE REVIEW

File: src/use_color_mode.rs [hunk 1]
```diff
@@ -255,22 +270,34 @@ fn get_store_signal(
...
+        let (store, set_store, remove) = use_storage_with_options::<ColorMode, FromToStringCodec>(
             storage,
             storage_key,
             UseStorageOptions::default()
                 .listen_to_storage_changes(listen_to_storage_changes)
                 .initial_value(initial_value),
         );
+        if update_previous_value {
+            remove();
+        }
```
Comment (mondeja) [inline src/use_color_mode.rs:290]: This behaviour of updating the value of storage with the url param could be parametrized with an option `initial_value_from_url_to_storage` or something else.

Comment (maccesch) [reply src/use_color_mode.rs:290]: 1. This is indeed not sth everybody might want. So yes please add an option for that.
2. removing the value here only to override any value down below seems redundant
3. please move the overriding code out of this function to make it more "pure"

**Review by maccesch — CHANGES_REQUESTED** [review]
Looks pretty good. I have added a few minor remarks

**Comment by mondeja** [comment]
Now looks much better.

**Comment by maccesch** [comment]
Nice work, thanks!

**Pull request merged**